gcc/rust/ast/rust-expr.h

```diff
@@ -2720,17 +2721,12 @@ class RangeExpr : public ExprWithoutBlock
 public:
   Location get_locus () const override final { return locus; }
 
-  // should never be called - error if called
-  void set_outer_attrs (std::vector<Attribute> /* new_attrs */) override
+  void set_outer_attrs (std::vector<Attribute> new_attrs) override
   {
-    rust_assert (false);
+    outer_attrs = std::move (new_attrs);
   }
 
-  std::vector<Attribute> &get_outer_attrs () override
-  {
-    // RangeExpr cannot have any outer attributes
-    rust_assert (false);
-  }
+  std::vector<Attribute> &get_outer_attrs () override { return outer_attrs; }
```

Comment:
I'm not sure about these changes - I don't think there's any way for `RangeExpr`s to have outer attributes at the moment

Reply:
Correct, see [the reference](https://doc.rust-lang.org/reference/expressions.html#expression-attributes).

Reply:
This only appears to apply to range expressions with a lower bound, rustc accepts the following
```rust
fn main() {
    #[cfg(all())]
    ..234;
}
``` 

That said, attribute parsing for expressions seems pretty complicated, so I've reverted the parsing to be as close as possible to the original handling before this patch.

Reply:
I wonder if that testcase passing is valid and not a bug - this is an expression, so we should definitely be hitting https://doc.rust-lang.org/stable/error_codes/E0658.html. But this is outside the scope of this PR :D

Reply:
to clarify, we should be hitting `error[[E0658]](https://doc.rust-lang.org/stable/error_codes/E0658.html): attributes on expressions are experimental`